pipeline/07-export.R

```diff
@@ -581,21 +575,46 @@ for (town in unique(assessment_pin_prepped$township_code)) {
         is.na(comp_pin_1_coord),
         NA,
         glue::glue(
-          '=HYPERLINK("[{workbook_name}]{comp_sheet_name}!',
-          '{comp_pin_1_coord}","{comp_pin_1}")'
+          '=HYPERLINK(@CELL("address",{comp_sheet_name}!{comp_pin_1_coord}),',
+          '"{comp_pin_1}")'
         )
       ),
       comp_pin_2 = ifelse(
         is.na(comp_pin_2_coord),
         NA,
         glue::glue(
-          '=HYPERLINK("[{workbook_name}]{comp_sheet_name}!',
-          '{comp_pin_2_coord}","{comp_pin_2}")'
+          '=HYPERLINK(@CELL("address",{comp_sheet_name}!{comp_pin_2_coord}),',
+          '"{comp_pin_2}")'
         )
       )
     ) %>%
     select(-ends_with("_coord"))
 
+  # Get range of rows in the PIN data + number of header rows
+  num_head <- 6 # Number of header rows
+  pin_row_range <- (num_head + 1):(nrow(assessment_pin_filtered) + num_head)
+  pin_row_range_w_header <- c(num_head, pin_row_range)
+  pin_col_range <- 1:60
+
+  # Calculate AVs so we can store them as separate, hidden columns for use
+  # in the neighborhood breakouts pivot table
+  assessment_pin_avs <- assessment_pin_filtered %>%
+    tibble::rowid_to_column("row_id") %>%
+    mutate(
+      row_id = row_id + num_head,
+      total_av = glue::glue("=S{row_id} * 0.1"),
+      av_difference = glue::glue("=(S{row_id} * 0.1) - (L{row_id} * 0.1)")
+    ) %>%
+    select(total_av, av_difference)
```

Comment:
I tried a couple different ways of computing these AVs for use in the breakout pivot table, largely constrained by the fact that [calculated fields](https://support.microsoft.com/en-us/office/calculate-values-in-a-pivottable-11f41417-da80-435c-a5c6-b0185e59da77) in pivot tables are not capable of any aggregation beyond summing columns in a range. I eventually decided to add hidden columns to the `PIN Detail` sheet containing formulas referencing the `Total` columns; the advantages of this approach are that:

* Formulas mean these values can dynamically update if a workbook user changes values in the `Total` column
* Keeping the columns in the `PIN Detail` sheet means that the pivot table only needs to point to one data source

Definitely open to feedback on this approach if you can think of a better way of doing it!

Reply:
I think this is totally fine. However, we should revisit this down the road to see if we can't make even more useful breakouts. Next season I suppose.